src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,36 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("User equals null");
+        }
+        final String login = user.getLogin();
+        final String password = user.getPassword();
+        final String usersId = String.valueOf(user.getId());
+        if (user.getAge() < 18) {
+            throw new RuntimeException("Invalid age");
+        }
+        if (password.length() < 6) {
+            throw new RuntimeException("The password is too short");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("There is already such login in the system");
+        }
+        if (!login.contains("@")) {
```

Comment:
I'm not sure if you need to additionally check the login for the presence of "@" and so on: [task](https://github.com/mate-academy/jv-registration-validation#registration-service)